---
output:
  md_document:
    variant: markdown_github
---
```{r setup, include = FALSE}
knitr::opts_chunk$set(comment = "")
```
# Simulated Data Summary, version 2
```{r, echo=FALSE}
cat(format(Sys.Date()))
```

The data was generated with a fastsimcoal simulation that created allele frequencies at equilibrium and were then used to initialize and run 3 generations of an rmetasim simulation.

## Scenario parameters

The scenarios were composed of the following parameter combinations:

```{r, echo=FALSE}
load("v2.params.rdata")
params[, 1:10]
```

The "island" model specifies a migration matrix such as the following from scenario 3, where the migration rate for a population is `r params$mig.rate[3]` split among the other 4 populations:

```{r, echo=FALSE}
params$mig.mat[3][[1]]
```

The "stepping.stone" model specifies a migration matrix such as the following from scenario 15, where the migration rate for a population is `r params$mig.rate[15]` split between the neighboring two populations:

```{r, echo=FALSE}
params$mig.mat[15][[1]]
```

## Files

All output files are contained in the folder "v#.sim.data". Each scenario has gtypes objects stored in a R workspace file, named "v#.gtypes.sc.rdata" where "sc" is the scenario number. This file contains two objects:

* `fsc.list` - A list of gtypes from fastsimcoal, one per replicate. The scenario parameters are stored as a one row data.frame in `attr(fsc.list, "params")`.
* `rms.list` - A list of gtypes from rmetasim after initialization with the corresponding gtypes object from `fsc.list`. This contains the final genotypes. The scenario parameters are also stored as a one row data.frame in `attr(rms.list, "params")`. 

## Diagnostics

```{r, include=FALSE}
library(tidyverse)
load("v2.summaries.rdata")
```
```{r, echo = FALSE}
pop.loc.df %>%
  filter(variable == "maf") %>%
  ggplot(aes(value)) +
  geom_histogram() +
  labs(x = "MAF", title = "MAF") +
  facet_grid(Nm ~ mig.type, labeller = label_both)

pop.struct.df %>%
  filter(metric == "est.fst") %>%
  mutate(ratio = log10(estimate / (1 / (4 * Nm + 1)))) %>%
  ggplot(aes(factor(mig.type), ratio)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_jitter(alpha = 0.6) +
  labs(
    x = "Migration Topology",
    y = expression(log[10]("estimated Fst / expected Fst (=1/(4Nm + 1)")),
    title = "Fst"
  ) +
  facet_wrap(~ Nm, ncol = 1, labeller = label_both)

ne.df %>%
  filter(variable == "Ne") %>%
  mutate(ratio = log10(value / Ne)) %>%
  ggplot(aes(factor(mig.type), ratio)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_jitter(alpha = 0.6) +
  labs(
    x = "Migration Topology",
    y = expression(log[10]("estimated Ne / simulation Ne")),
    title = "ldNe"
  ) +
  facet_wrap(~ Nm, ncol = 1, labeller = label_both)

```

